From 52d875155183817e4a5b1e13c683301d50d2bda1 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 10 Nov 2022 13:40:52 +0200
Subject: [PATCH] [SW-105079] gaudi3: add pll vco enable support

In addition to current pll configuration, VCO configuration needs
to be done for gaudi3. This patch add function to enable
VCO as part of hl_pll_set routine.

Change-Id: I6e924ef2ce0a62f7be79b3aee14b2840d69d9e75
---

diff --git a/arch/gaudi3/common/pll.c b/arch/gaudi3/common/pll.c
index 6d4966b..979e955 100644
--- a/arch/gaudi3/common/pll.c
+++ b/arch/gaudi3/common/pll.c
@@ -77,3 +77,12 @@
 	regmap_writel(1, cfg->base + GAUDI3_PLL_DIV_SEL_DELAY);
 }
 
+#define VCO_CTRL_FOUTVCOEN_M	0x10
+void hl_pll_vco_en(struct hl_pll_cfg *cfg)
+{
+	u32 val;
+
+	val = regmap_readl(cfg->base + GAUDI3_PLL_VCO_CTRL);
+	val |= VCO_CTRL_FOUTVCOEN_M;
+	regmap_writel(val, cfg->base + GAUDI3_PLL_VCO_CTRL);
+}
diff --git a/lib/common/pll.c b/lib/common/pll.c
index 30159e7..a14f60c 100644
--- a/lib/common/pll.c
+++ b/lib/common/pll.c
@@ -305,6 +305,7 @@
 }
 
 __weak void hl_pll_set_div_sel_delay(struct hl_pll_cfg *cfg) { }
+__weak void hl_pll_vco_en(struct hl_pll_cfg *cfg) {}
 
 static void hl_pll_set(struct hl_pll_cfg *cfg)
 {
@@ -320,6 +321,8 @@
 
 	hl_pll_div_init(cfg);
 
+	hl_pll_vco_en(cfg);
+
 	/* to be on a safe side, poll on the PLL lock status bit */
 	is_locked = hl_pll_wait_for_lock(cfg);
 
